source-code/sdk-js/src/helpers/ast.ts

```diff
@@ -78,31 +83,37 @@ export const wrapVariableDeclaration = (async (sourceAst, searchIdentifier, iden
  * @returns Promise that either resolves to the resulting ast or throws if the given insertion position couldn't be found
  * or the positional information array length does not satisfy length % 3 === 2
  */
-export const insertAst = (async (sourceAst, ast, { before, after }) => {
-	const sourceAstClone = structuredClone(sourceAst)
-	const position = after || before
-	if (position.length % 3 !== 2) {
-		throw new Error("The length of 'before' or 'after' has to be a multiple of two.")
-	}
-	const targetParent = get(sourceAstClone, position.slice(0, -2), sourceAstClone)
-	const targetArrayKey = position.at(-2)
-	const targetArrayIndex = Number(position.at(-1))
-	let newAst
-	walk(sourceAstClone, {
-		enter(node) {
-			const newNode = structuredClone(node)
-			if (node === targetParent) {
-				const targetArray = get(newNode, targetArrayKey as string, undefined) as Node[] | undefined
-				if (!targetArray) throw new Error("Couldn't access given position in AST.")
-				else targetArray.splice(targetArrayIndex + (before ? 0 : 1), 0, structuredClone(ast))
-				// this.replace doesn't work on the root node - that's why we need the next line
-				if (node === sourceAstClone) {
-					newAst = newNode as Program
+export const insertAst = ((sourceAst, ast, { before, after }) => {
+	try {
+		const sourceAstClone = structuredClone(sourceAst)
+		const position = after || before
+		if (position.length % 3 !== 2) {
+			throw new Error("The length of 'before' or 'after' has to be a multiple of two.")
+		}
+		const targetParent = get(sourceAstClone, position.slice(0, -2), sourceAstClone)
+		const targetArrayKey = position.at(-2)
+		const targetArrayIndex = Number(position.at(-1))
+		let newAst
+		walk(sourceAstClone, {
+			enter(node) {
+				const newNode = structuredClone(node)
+				if (node === targetParent) {
+					const targetArray = get(newNode, targetArrayKey as string, undefined) as
+						| Node[]
+						| undefined
+					if (!targetArray) throw new Error("Couldn't access given position in AST.")
+					else targetArray.splice(targetArrayIndex + (before ? 0 : 1), 0, structuredClone(ast))
+					// this.replace doesn't work on the root node - that's why we need the next line
+					if (node === sourceAstClone) {
+						newAst = newNode as Program
+					}
+					this.replace(newNode)
+					this.skip()
 				}
-				this.replace(newNode)
-				this.skip()
-			}
-		},
-	})
-	return newAst || sourceAstClone
-}) satisfies insertAst
+			},
+		})
+		return [newAst || sourceAstClone, undefined]
+	} catch (error) {
+		return [undefined, error as Error]
```

Comment:
same as above

Reply:
Keeping this, as `walk`can throw.

Reply:
Then please return a custom Error class here too. So we know what kind of error happened